### PR TITLE
2360 Enhance the data type of GradCam

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2360 .

### Description
This PR enhanced the return data type/device of `GradCam` to be consistent with input data.
It follows the regular usages in user's feedback and our tutorial:
https://github.com/Project-MONAI/tutorials/blob/master/modules/interpretability/covid_classification.ipynb
And it's also consistent with our new feature that updates all the transforms to support input/output Tensor data.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by rijobro — APPROVED** [review]
thanks

File: monai/visualize/class_activation_maps.py [hunk 1]
```diff
@@ -17,24 +17,30 @@
...
 from monai.visualize.visualizer import default_upsampler
 
 __all__ = ["CAM", "GradCAM", "GradCAMpp", "ModelWithHooks", "default_normalizer"]
 
 
-def default_normalizer(x) -> np.ndarray:
+def default_normalizer(x: NdarrayTensor) -> NdarrayTensor:
     """
     A linear intensity scaling by mapping the (min, max) to (1, 0).
+    The output data type and device will be consistent with input data.
```
Comment (wyli) [inline monai/visualize/class_activation_maps.py:32]: For the docstring, I don't think the data type should always be consistent here, input could be integers and output floats. Probably 'N.B.' is not a typo

Comment (Nic-Ma) [reply monai/visualize/class_activation_maps.py:32]: OK, that makes sense, I will update it.

Thanks.

Comment (Nic-Ma) [reply monai/visualize/class_activation_maps.py:33]: updated in the latest commit.
Thanks.

**Pull request merged**